### PR TITLE
chore: phase out ethers rpc block types

### LINKED ISSUE #1439: Phase out rpc::BlockNumber with BlockNumberOrTag
### Describe the feature

Followup of #1237

### Additional context

[Request for a followup](https://github.com/paradigmxyz/reth/pull/1237#pullrequestreview-1300418569)

### PR DESCRIPTION
**Pull request by mattsse**
closes https://github.com/paradigmxyz/reth/issues/1439
supersedes https://github.com/paradigmxyz/reth/pull/1441

replace 
* `ethers::rpc::BlockId` with `reth::BlockId`
* `ethers::rpc::BlockNumber` with `reth::BlockNumberOrTag`

### CODE REVIEW

File: crates/rpc/rpc-api/src/eth.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ pub trait EthApi {
     async fn accounts(&self) -> Result<Vec<Address>>;
 
     /// Returns the number of most recent block.
-    #[method(name = "eth_blockNumber")]
+    #[method(name = "eth_BlockNumberOrTag")]
```
Comment (rkrasiuk) [inline crates/rpc/rpc-api/src/eth.rs:33]: replace all went a bit too far 😄. this shouldn't change

File: crates/rpc/rpc-api/src/eth.rs [hunk 2]
```diff
@@ -43,23 +43,30 @@ pub trait EthApi {
...
+    ) -> Result<Option<U256>>;
 
     /// Returns the number of uncles in a block from a block matching the given block hash.
     #[method(name = "eth_getUncleCountByBlockHash")]
     async fn block_uncles_count_by_hash(&self, hash: H256) -> Result<U256>;
 
     /// Returns the number of uncles in a block with given block number.
-    #[method(name = "eth_getUncleCountByBlockNumber")]
-    async fn block_uncles_count_by_number(&self, number: BlockNumber) -> Result<U256>;
+    #[method(name = "eth_getUncleCountByBlockNumberOrTag")]
```
Comment (rkrasiuk) [inline crates/rpc/rpc-api/src/eth.rs:68]: same here

File: crates/rpc/rpc-api/src/eth.rs [hunk 3]
```diff
@@ -70,10 +77,10 @@ pub trait EthApi {
     ) -> Result<Option<RichBlock>>;
 
     /// Returns an uncle block of the given block and index.
-    #[method(name = "eth_getUncleByBlockNumberAndIndex")]
+    #[method(name = "eth_getUncleByBlockNumberOrTagAndIndex")]
```
Comment (rkrasiuk) [inline crates/rpc/rpc-api/src/eth.rs:80]: and here

File: crates/rpc/rpc-api/src/eth.rs [hunk 4]
```diff
@@ -90,10 +97,10 @@ pub trait EthApi {
     ) -> Result<Option<Transaction>>;
 
     /// Returns information about a transaction by block number and transaction index position.
-    #[method(name = "eth_getTransactionByBlockNumberAndIndex")]
+    #[method(name = "eth_getTransactionByBlockNumberOrTagAndIndex")]
```
Comment (rkrasiuk) [inline crates/rpc/rpc-api/src/eth.rs:100]: and here

**Review by rkrasiuk — CHANGES_REQUESTED** [review]

**Review by rkrasiuk — APPROVED** [review]

**Review by gakonst — COMMENTED** [review]
amazing

**Pull request merged**